src/CMakeLists.txt

```diff
@@ -1,14 +1,17 @@
 
-add_subdirectory("main/c")
-add_subdirectory("main/web")
+add_subdirectory(main/c)
 
-if (SEASOCKS_EXAMPLE_APP)
-  add_subdirectory("app/c")
-endif ()
+if(SEASOCKS_SHARED)
+    add_subdirectory(main/web)
+
+    if (SEASOCKS_EXAMPLE_APP)
```

Comment:
The cmake script currently builds both shared and static libraries, which can be disabled.
Because the examples+tests link to the shared library, this dependency might be missing
I'll take another look at this and add a new option: `SEASOCKS_APPS_USE_SHARED`.

Reply:
No need to make things more complex: I was just trying to understand :)

Reply:
Still, this is a consequence of not using cmake's `BUILD_SHARED_LIBS`.
Because right now, the static library is built, but not used/tested anywhere.
This is a potential bug.

Reply:
Thanks @madebr

Reply:
What option do you prefer?
- [ ]  Add a `SEASOCKS_APPS_USE_SHARED` that lets the user choose to what type of library the examples/tests link. This has the potential for people forgetting to test linking to the static library.
- [ ] Add the examples/tests twice.
    The examples/tests can be built once, but linked twice.

Reply:
I'm not worried either way: I doubt there are many categories of errors which mean that static libs work when dynamic don't. That said; if it's not much work then I'd prefer the second option, but the first works too if the CI system matrix is expanded to build with it on and off.

Reply:
I've applied the second approach done this in the latest commit.
Most of the additional complexity comes from us building both shared and static libraries (as opposed to using cmake's `BUILD_SHARED_LIBS`).

When configuring cmake as `cmake .. -DCMAKE_POSITION_INDEPENDENT_CODE=ON`, the number of source compilatations should be equal, but the number of linkings should be doubled.

Reply:
Thanks!